src/main/java/graphql/servlet/GraphQLQueryProvider.java

```diff
@@ -14,11 +14,37 @@
  */
 package graphql.servlet;
 
+import graphql.schema.GraphQLFieldDefinition;
 import graphql.schema.GraphQLObjectType;
 
+import java.util.Collection;
+
+/**
+ * This interface is used by OSGi bundles to plugin new field into the root query type
+ */
 public interface GraphQLQueryProvider {
-    GraphQLObjectType getQuery();
-    Object context();
+
+    /**
+     * @return a collection of field definitions that will be added to the root query type.
+     */
+    default Collection<GraphQLFieldDefinition> getQueryFieldDefinitions() { return null; }
+
+    /**
+     * @deprecated use query field definitions instead
+     * @return a GraphQL object type to add to the root query type
+     */
+    default GraphQLObjectType getQuery() { return null; }
```

Comment:
Per my comment, I don't think there's much need to preserve this as deprecated right now.  Once this is removed I'll merge this change.